### PR TITLE
John --> master: documentation sync, slight argument changes

### PR DESCRIPTION
**Pull request by jhallard**
Improved / updated documentation slightly, added a feature to pass in arbitrary meta-data to the hooks module. Also a bug-fix or two my @mdipierro

### CODE REVIEW

**Comment by jhallard** [comment]
also I have a note to change the `test.camio.com` back to production once everything is deployed. leaving it at `test.camio.com` right now while I continue finalizing stuff.

File: batch_import/camio_hooks.py [hunk 1]
```diff
@@ -70,6 +76,7 @@ def get_camera_plan():
                 CAMIO_PARAMS['plan'], [CAMIO_PLANS[key] for key in CAMIO_PLANS])
         return CAMIO_PLANS['plus'] 
     return CAMIO_PLANS.get(CAMIO_PARAMS['plan'].lower())
+>>>>>>> 87878f481af1721d071c5dd44ae469d34de5852d
```
Comment (jhallard) [inline batch_import/camio_hooks.py]: @mdipierro merge conflict in this file

**Comment by jhallard** [comment]
@jcmaslan @mdipierro this is locked from me merging, can you guys approve?

File: batch_import/batch_download.py [hunk 2]
```diff
@@ -217,28 +217,50 @@
...
-	these search results can then be parsed and the meta-data about the labels added to each event can be extracted and assembled
-	into a dictionary of some sorts to be returned to the user
-	"""
-	pass
+
+
+class BatchDownloader(object):
+
+    def __init__(self):
+        # TODO - change the URLs to test.camio.com instead of test.camio.com after deployed to prod
```
Comment (jcmaslan) [inline batch_import/batch_download.py]: www

File: batch_import/batch_download.py [hunk 3]
```diff
@@ -217,28 +217,50 @@
...
+        # positional args
+        self.parser.add_argument('job_id', type=str, help='the ID of the job that you wish to download the labels for')
+        # optional arguments
+        self.parser.add_argument('-a', '--access_token', type=str, help='your Camio OAuth token (if not given we check the CAMIO_OAUTH_TOKEN envvar)')
+
+    def get_access_token(self):
+        if not self.CAMIO_PARAMS.get('access_token'):
+            token = os.environ.get(self.CAMIO_OAUTH_TOKEN_ENVVAR)
+            if not token:
+                fail("unable to find Camio OAuth token in either hook-params json or CAMIO_OAUTH_TOKEN envvar. \
```
Comment (jcmaslan) [inline batch_import/batch_download.py]: access_token (make error message concrete about what's missing)

File: batch_import/camio_hooks.py [hunk 4]
```diff
@@ -6,28 +6,85 @@
...
+Log = None
+
+# plan definitions for actual_values entry
+CAMIO_PLANS = { 'pro': 'PRO', 'plus': 'PLUS', 'basic': 'BASIC' }
+
+def fail(msg, *args):
+    Log.error(msg, *args)
+    sys.exit(1)
+
+def set_hook_data(data_dict):
```
Comment (jcmaslan) [inline batch_import/camio_hooks.py:42]: Isn't this any json of the caller's choice?

File: batch_import/camio_hooks.py [hunk 5]
```diff
@@ -6,28 +6,85 @@
...
+    date = datetime.datetime.strptime(timestamp[:23], "%Y-%m-%dT%H:%M:%S.%f")
+    date = date + datetime.timedelta(seconds=seconds)
+    return date.isoformat()
 
 def get_device_id():
-    return os.environ.get(CAMIO_BOX_DEVICE_ID_ENVVAR) 
+    if not CAMIO_PARAMS.get('device_id'):
+        device = os.environ.get(CAMIO_BOX_DEVICE_ID_ENVVAR)
+        if not device:
+            fail("unable to find Camio Box Device ID in either hook-params json or CAMIO_BOX_DEVICE_ID_ENVVAR envvar.\
```
Comment (jcmaslan) [inline batch_import/camio_hooks.py]: device_id

File: batch_import/camio_hooks.py [hunk 6]
```diff
@@ -6,28 +6,85 @@
...
+def dateshift(timestamp, seconds):
+    date = datetime.datetime.strptime(timestamp[:23], "%Y-%m-%dT%H:%M:%S.%f")
+    date = date + datetime.timedelta(seconds=seconds)
+    return date.isoformat()
 
 def get_device_id():
-    return os.environ.get(CAMIO_BOX_DEVICE_ID_ENVVAR) 
+    if not CAMIO_PARAMS.get('device_id'):
+        device = os.environ.get(CAMIO_BOX_DEVICE_ID_ENVVAR)
+        if not device:
```
Comment (jcmaslan) [inline batch_import/camio_hooks.py]: device_id

File: batch_import/camio_hooks.py [hunk 7]
```diff
@@ -52,20 +109,23 @@ def get_device_data(host, port):
...
+    return response[local_camera_id]
 
 def register_camera(camera_name, host=None, port=None):
     """
     arguments:
         camera_name   - the name of the camera (as parsed from the filename) 
         host          - the URI/IP address of the segmenter being used
         port          - the port to access the webserver of the segmenter
-    returns: this function returns the Camio-specific camera ID.
+    returns: this function returns a dctionary describing the new camera, including the camera ID
```
Comment (jcmaslan) [inline batch_import/camio_hooks.py:163]: not dictionary

File: batch_import/camio_hooks.py [hunk 8]
```diff
@@ -113,9 +177,10 @@ def post_video_content(host, port, camera_name, camera_id, filepath, timestamp,
                  this function where the logic should exist to post the file content to a video
                  segmenter.
     """
-    filehash = None
     device_id = CAMIO_PARAMS.get('device_id')
-    if not device_id: return False
+    if not device_id: 
+        Log.error("unable to find Camio Box device ID in --hook_data_json or env-vars.")
```
Comment (jcmaslan) [inline batch_import/camio_hooks.py:226]: device_id

File: batch_import/camio_hooks.py [hunk 9]
```diff
@@ -155,29 +236,50 @@ def assign_job_ids(self, db, unscheduled):
             n += shard['item_count']
             upload_urls.append((n, shard_id, shard['upload_url']))
 
+        # for debug only 
+        for item in upload_urls:
+            Log.debug("upload item: %r", item[0])
```
Comment (jcmaslan) [inline batch_import/camio_hooks.py:298]: Maybe we say import item since not all are uploaded.

File: batch_import/camio_hooks.py [hunk 10]
```diff
@@ -155,29 +236,50 @@ def assign_job_ids(self, db, unscheduled):
...
+            'item_count':len(rows),
+            'hash_map': hash_map
+            }
+        Log.debug("registering job:\n ID=%s\n shard ID=%s\n num items=%d\n file-map=%r", 
+                job_id, shard_id, len(rows), hash_map)
         url = rows[0]['upload_url']
-        requests.put(url, json=payload)
+        ret = requests.put(url, json=payload)
+        if not ret.status_code in (200, 204):
+            Log.error("error registering job: %s", job_id)
```
Comment (jcmaslan) [inline batch_import/camio_hooks.py:338]: job_id

**Review by jcmaslan — APPROVED** [review]
Minor changes requested

File: batch_import/batch_download.py (hunk 2 above)
Comment (jhallard) [reply batch_import/batch_download.py]: global search and replace bit me.. good catch

File: batch_import/camio_hooks.py (hunk 4 above)
Comment (jhallard) [reply batch_import/camio_hooks.py:42]: it's been de-serialized into a python dictionary by the time it's passed in

Comment (jcmaslan) [reply batch_import/camio_hooks.py:42]: But can't a caller pass a list of they want?

**Pull request merged**